src/__tests__/waitFor.test.tsx

```diff
@@ -78,6 +78,38 @@ test('waits for element with custom interval', async () => {
   expect(mockFn).toHaveBeenCalledTimes(2);
 });
 
+test('waits for async event with fireEvent', async () => {
+  const Comp = ({ onPress }: { onPress: () => void }) => {
+    const [state, setState] = React.useState(false);
+
+    React.useEffect(() => {
+      if (state) {
+        onPress();
+      }
+    }, [state, onPress]);
+
+    return (
+      <Pressable
+        onPress={async () => {
+          await Promise.resolve();
+          setState(true);
```

Comment:
Would it matter if we invoked `onPress` here after `await Promise` instead of in `useEffect`?

Reply:
Ok, I understand. Pls add some comments that describes this, as without that knowledge the code looks very unnatural and badly written event handling